### PR TITLE
Dataset generation

### LINKED ISSUE #2: Extend TOFU Dataset
- We would like to extend the TOFU dataset in the following directions:
  - Amount of data
  - Different levels of granularity
  - Different levels of interconnection
- Implement functions to generate additional TOFU data deterministically given a seed number (or create separate dataset?)
- As in #1, we want to be able to manage these for creating retain/forget sets later on

### PR DESCRIPTION
**Pull request by jack89roberts**

### CODE REVIEW

File: src/arcsf/data/generation/name_bank/publisher_names.csv [hunk 1]
```diff
@@ -0,0 +1,40 @@
+Albatross Press
+Clive & Sons
+Pelican Publishing
```
Comment (jack89roberts) [inline src/arcsf/data/generation/name_bank/publisher_names.csv]: to rename here too?

File: src/arcsf/data/generation/utils.py [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+
+    Returns:
+        extracted string
+    """
+    return string.split(start)[1].split(end)[0]
+
+
+def str_time_prop(start, end, time_format, diff):
+    """
+    Get a time at a proportion of a range of two formatted times.
```
Comment (jack89roberts) [inline src/arcsf/data/generation/utils.py]: docstring is talking about `prop` but looks like the function has been changed

File: src/arcsf/data/generation/scripts/generate_tofu.py [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+from arcsf.data.generation.utils import (
+    AuthorSampler,
+    BookSampler,
+    Formatter,
+    PublisherSampler,
+    flatten,
+)
+
+# RANDOM SEED
+# random.seed(42)
```
Comment (jack89roberts) [inline src/arcsf/data/generation/scripts/generate_tofu.py:28]: do we know what seed the dataset was generated with?

Comment (jack89roberts) [reply src/arcsf/data/generation/scripts/generate_tofu.py:28]: seems not

File: src/arcsf/data/generation/name_bank/publisher_names.csv (hunk 1 above)
Comment (J-Dymond) [reply src/arcsf/data/generation/name_bank/publisher_names.csv]: done

File: src/arcsf/data/generation/utils.py (hunk 2 above)
Comment (J-Dymond) [reply src/arcsf/data/generation/utils.py]: quickly changed this

**Review by J-Dymond — COMMENTED** [review]
Briefly discussed code with JR. Appropriate to merge now.

**Review by J-Dymond — APPROVED** [review]
Briefly discussed code with JR. Appropriate to merge now. This time an approving review...

**Pull request merged**